src/middleware.js

```diff
@@ -23,10 +23,10 @@ export const createOfflineMiddleware = (config: Config) => (store: any) => (
 
   // create promise to return on enqueue offline action
   if (action.meta && action.meta.offline) {
-    promise = registerAction(action.meta.transaction);
+    promise = registerAction(offlineAction.meta.transaction);
```

Comment:
Thought I had written enough tests, but I suppose that's what I get for not trying it in an actual app. Regardless this isn't going to work either.

The first action in the outbox, `offlineAction`, isn't necessarily the current stamped action. We are after the most recently added action, which would be the last one. But since we are only interested in the transaction id, we could just use `lastTransaction`.

In fact, since we now allow the user to control adding to and reading from the outbox, I think this is what we will have to do. Unless we want to add another method to the queue config.

Reply:
Yes just realized that this wasn't always working. Thanks for pointing me in the right direction. 😄